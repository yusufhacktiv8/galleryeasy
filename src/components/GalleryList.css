.grid {
  display: flex;
  flex-flow: row wrap;
  width: 80%;
  margin-left: auto;
  margin-right: auto;
}
