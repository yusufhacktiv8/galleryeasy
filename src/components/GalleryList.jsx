import React from 'react';
import GalleryListItem from './GalleryListItem';

export default ({ items }) => {
  return (
    <div className="container">
      <div className="row">
        {
          items.map(item => (
            <div className="col-3" key={item.id}>
              <GalleryListItem url={item.url} favourited={item.favourited} />
            </div>
          ))
        }
      </div>
    </div>
  );
};
