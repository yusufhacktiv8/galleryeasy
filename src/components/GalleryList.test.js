import React from 'react';
import renderer from 'react-test-renderer';
import GalleryList from './GalleryList';

test('GalleryList correctly rendered', () => {
  const items = [
    {
      id: 1,
      url: '/test1.png',
      favourited: true,
    },
    {
      id: 2,
      url: '/test2.png',
      favourited: false,
    },
  ];
  const component = renderer.create(
    <GalleryList items={items} />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
