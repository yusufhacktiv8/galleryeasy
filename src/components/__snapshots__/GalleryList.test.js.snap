// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleryList correctly rendered 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-3"
    >
      <div
        className="gallery-list-item"
        onMouseEnter={[Function]}
        onMouseOut={[Function]}
        role="link"
      >
        <img
          url="/test1.png"
        />
        <img
          alt="Tag"
          className=""
          onClick={[Function]}
          src="/images/tagimage.png"
        />
      </div>
    </div>
    <div
      className="col-3"
    >
      <div
        className="gallery-list-item"
        onMouseEnter={[Function]}
        onMouseOut={[Function]}
        role="link"
      >
        <img
          url="/test2.png"
        />
      </div>
    </div>
  </div>
</div>
`;
